File: venv_11_3_1/share/doc/gmsh/examples/api/onelab_run_auto.cpp
#include <iostream>
#include <gmsh.h>

int main(int argc, char **argv)
{
  if(argc < 2) {
    std::cout << "Usage: " << argv[0] << " file [options]" << std::endl;
    return 0;
  }

  gmsh::initialize();

  gmsh::open(argv[1]);

  gmsh::onelab::run();

  std::string json;
  gmsh::onelab::get(json);
  std::cout << json;

  gmsh::finalize();

  return 0;
}

File: venv_11_3_1/share/doc/gmsh/examples/api/onelab_run_auto.py
import gmsh
import sys

# This shows how ONELAB clients that communicate with the ONELAB server through
# sockets can be executed using the Gmsh API, using client information stored in
# the "Solver.Name?", "Solver.Extension?" and ""Solver.Executable?" Gmsh options

# By default Gmsh defines Solver.Name0 = "GetDP" and Solver.Extension0 = ".pro".
# Provided that Solver.Executable0 points to the GetDP executable, if you open a
# file with the ".pro" extension with gmsh.open(), then call gmsh.onelab.run(),
# the getdp solver will be run.

# If Solver.Executable0 is not set, you can set it using
#
# gmsh.option.setString('Solver.Executable0', '/path/to/getdp')

if len(sys.argv) < 2:
    print("Usage: " + sys.argv[0] + " file [options]")
    exit(0)

gmsh.initialize()

gmsh.open(sys.argv[1])

# attempts to run a client selected when opening the file (e.g. a .pro file)
gmsh.onelab.run()

json = gmsh.onelab.get()
print(json)

gmsh.finalize()
